controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -222,29 +323,49 @@ func (r *StorageConsumerReconciler) own(resource metav1.Object) error {
 func (r *StorageConsumerReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	enqueueStorageConsumerRequest := handler.EnqueueRequestsFromMapFunc(
 		func(context context.Context, obj client.Object) []reconcile.Request {
-			labels := obj.GetLabels()
-			if value, ok := labels[StorageConsumerNameLabel]; ok {
-				return []reconcile.Request{{
+			// Get the StorageConsumer objects
+			consumers := &ocsv1alpha1.StorageConsumerList{}
+			err := r.Client.List(context, consumers, &client.ListOptions{Namespace: obj.GetNamespace()})
+			if err != nil {
+				r.Log.Error(err, "Unable to list StorageConsumers")
+				return []reconcile.Request{}
+			}
+
+			// Return name and namespace of the StorageClusters object
+			request := []reconcile.Request{}
+			for _, sc := range consumers.Items {
+				request = append(request, reconcile.Request{
 					NamespacedName: types.NamespacedName{
-						Name:      value,
-						Namespace: obj.GetNamespace(),
+						Namespace: sc.Namespace,
+						Name:      sc.Name,
 					},
-				}}
+				})
 			}
-			return []reconcile.Request{}
+
+			return request
 		},
 	)
 
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&ocsv1alpha1.StorageConsumer{}, builder.WithPredicates(
-			predicate.GenerationChangedPredicate{},
-		)).
+		For(&ocsv1alpha1.StorageConsumer{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
 		Owns(&nbv1.NooBaaAccount{}).
-		// Watch non-owned resources cephBlockPool
-		// Whenever their is new cephBockPool created to keep storageConsumer up to date.
-		Watches(&rookCephv1.CephBlockPool{},
-			enqueueStorageConsumerRequest,
+		Owns(
+			&rookCephv1.CephBlockPoolRadosNamespace{},
+			builder.MatchEveryOwner,
+			builder.WithPredicates(
+				predicate.GenerationChangedPredicate{},
+			),
+		).
+		Owns(
+			&rookCephv1.CephFilesystemSubVolumeGroup{},
+			builder.MatchEveryOwner,
+			builder.WithPredicates(
+				predicate.GenerationChangedPredicate{},
+			),
 		).
+		// Watch non-owned resources
+		Watches(&rookCephv1.CephBlockPool{}, enqueueStorageConsumerRequest).
+		Watches(&rookCephv1.CephFilesystem{}, enqueueStorageConsumerRequest).
```

Comment:
Why this kind of watch and not Owns watch?

Reply:
StorageCluster is the owner of the CephBlockPool and CephFilesystem

Reply:
Ok in that case we do we need to watch the cephfilesystem ?